src/F_LSM6DSL.hpp

```diff
@@ -136,5 +136,17 @@ class LSM6DSL : public IMUBase {
 			dest[i++] = Wire.read();
 		}         // Put read results in the Rx buffer
 	}
+
+	uint8_t checkReady(uint8_t address, uint8_t timeout)
+	{
+		uint8_t IMUWhoAmI = 0;
+		// Wait until a valid byte is returned, up until timeout value.
+		for (uint8_t checkCount = timeout; checkCount > 0; checkCount--) {
+			IMUWhoAmI = readByte(address, LSM6DSL_WHO_AM_I);
+			if (IMUWhoAmI == 0xFF) { delay(1); } else { break; }
```

Comment:
I'm actually not sure why `0xFF` is returned. I think that what might be actually getting returned by `Wire.read()` is `-1`, but since `readByte()` returns an unsigned integer, I think the value gets wrapped around to 255 instead of -1. A bigger set of changes could make this more logical and check for a `-1` value, but it would require changing many parameters and returns types from `uint8_t` to `int8_t`.

Reply:
I tried changing `readByte()` to return `int8_t` and even after that the "not ready" return response was `0xFF`. So I guess this isn't an unsigned integer problem. That means that even if `readByte()` were changed to return signed values, this logic should still work.